esp-hal/src/rmt.rs

```diff
@@ -978,6 +984,15 @@ where
     phantom: PhantomData<M>,
 }
 
+impl<M: crate::Mode, const CHANNEL: u8> Drop for Channel<M, CHANNEL> {
+    fn drop(&mut self) {
+        REF_COUNTER.fetch_sub(1, Relaxed);
+        if REF_COUNTER.load(Relaxed) == 0 {
```

Comment:
```suggestion
        if REF_COUNTER.fetch_sub(1, Relaxed) == 1 {
```

There's no reason to read one more time.

Reply:
This Drop also means `new_internal` should no longer enable clocks. Otherwise it's possible to create Rmt (enables clocks), then drop it (it won't disable clocks because no Channels have been created).

Reply:
Yes, this is a failed try for peripherals that has Channels/Operators etc. We come up with this idea with @bjoernQ but yeah, it's not quite good yet. I just wanted to keep it here (for now) to show what kind of issues we have to solve (or think of).